units/commit_queue.go

```diff
@@ -289,10 +289,9 @@ func (j *commitQueueJob) TryUnstick(ctx context.Context, cq *commitqueue.CommitQ
 				"source":  "commit queue",
 				"job_id":  j.ID(),
 			})
-			j.AddError(model.SetActiveState(mergeTask, evergreen.APIServerTaskActivator, true))
```

Comment:
I made these checks just log since they can be racy

Reply:
The point of this job originally was to actually unstick the queue if something was wrong. Is it worth having a job that only adds to logs at the error level? Could we instead log the response of SetActiveState at a low level rather than add it as an error to the job? I assume the race you're encountering is if the task has since been activated. Or I think this job used to only do something if the patch had been at the top of the queue for a certain amount of time; you could consider adding something like that in.

Reply:
It's more that the race is with a task failing and causing the entry to be removed right after this would run. All 3 of these things will momentarily be true when that happens, it's just bad if it's permanently like that. I'm going to periodically check the logging and see if these are deterministic enough to be able to automatically fix them

Reply:
Gotcha-- it makes sense if this is an in-progress thing we'd like to think about, as long as we don't intend for this job to always be errors only. Can you link a ticket for re-visiting the job so it doesn't slip through the cracks?

Reply:
yep made https://jira.mongodb.org/browse/EVG-14316